sematic/resolvers/resource_requirements.py

```diff
@@ -194,6 +194,55 @@ def __post_init__(self):
             )
 
 
+@dataclass
+class KubernetesCapabilities:
+    """Capabilities associated with a Kubernetes Security Context.
+
+    For more docs, see:
+    https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#capabilities-v1-core
+
+    Attributes
+    ----------
+    add:
+        Added capabilities
+    drop:
+        Dropped capabilities
+    """
+
+    add: List[str] = field(default_factory=list)
+    drop: List[str] = field(default_factory=list)
+
+
+@dataclass
+class KubernetesSecurityContext:
+    """A security context the Sematic job should run with.
+
+    Docs sourced from:
+    https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#securitycontext-v1-core
+
+    Attributes
+    ----------
+    allow_privilege_escalation:
+        AllowPrivilegeEscalation controls whether a process can gain more privileges
+        than its parent process. This bool directly controls if the no_new_privs
+        flag will be set on the container process. AllowPrivilegeEscalation is true
+        always when the container is: 1) run as Privileged 2) has CAP_SYS_ADMIN Note
+        that this field cannot be set when spec.os.name is windows.
+    privileged:
+        Run container in privileged mode. Processes in privileged containers are
+        essentially equivalent to root on the host. Defaults to false. Note that
+        this field cannot be set when spec.os.name is windows.
+    capabilities:
+        The capabilities to add/drop when running containers. Defaults to the default
+        set of capabilities granted by the container runtime. Note that this field
+        cannot be set when spec.os.name is windows.
+    """
+
+    allow_privilege_escalation: bool
```

Comment:
huh do you need both here? like i'd think `privileged` implies `allow_privilege_escalation`.  at least when i look at a pod that does have CAP_SYS / FUSE working, it looks like this:

```
securityContext:
            privileged: true
            capabilities:
              add:
                # SYS_ADMIN is needed for run `mount` command in the container
                - SYS_ADMIN
             ```

i am no k8s expert here tho, just reporting was i was trying to emulate

Reply:
this API just mirrors the [configuration API of K8s](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#securitycontext-v1-core), which has both `allowPrivilegeEscalation` and `privileged`